exchanges/btcmarkets/btcmarkets.go

```diff
@@ -379,52 +553,55 @@ func (b *BTCMarkets) SendHTTPRequest(path string, result interface{}) error {
 }
 
 // SendAuthenticatedRequest sends an authenticated HTTP request
-func (b *BTCMarkets) SendAuthenticatedRequest(reqType, path string, data, result interface{}) (err error) {
+func (b *BTCMarkets) SendAuthenticatedRequest(method, path string, data map[string]interface{}, result interface{}, specialCase []interface{}) (err error) {
 	if !b.AllowAuthenticatedRequest() {
 		return fmt.Errorf(exchange.WarningAuthenticatedRequestWithoutCredentialsSet,
 			b.Name)
 	}
 
-	n := b.Requester.GetNonce(true).String()[0:13]
-
-	var req string
-	payload := []byte("")
+	strTime := strconv.FormatInt(time.Now().UTC().UnixNano()/1000000, 10)
 
-	if data != nil {
+	var body io.Reader
+	var payload, hmac []byte
+	switch {
+	case len(data) != 0:
```

Comment:
ultra-lame niterino: length can never be less than 0, so can change `len(data) > 0`